src/main/java/de/tum/in/www1/artemis/domain/Rating.java

```diff
@@ -0,0 +1,71 @@
+package de.tum.in.www1.artemis.domain;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.MapsId;
+import javax.persistence.OneToOne;
+import javax.persistence.Table;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+
+/**
+ * A Rating.
+ */
+@Entity
+@Table(name = "result_rating")
+@JsonInclude(Include.NON_EMPTY)
+public class Rating implements Serializable {
+
+    @Id
+    private Long id;
+
+    @Column(name = "rating")
+    private Integer rating;
+
+    @MapsId
+    @OneToOne(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
```

Comment:
I don’t think cascade makes sense here. This would mean a result is deleted when a rating is deleted

Reply:
From my understanding Cascade.ALL propagates all operations — including Hibernate-specific ones — from a parent to a child entity. In an uni-directional 1-1 mapping using "MapsId", the referenced table is the parent entity and the referencing one is the child entity. Therefore I would delete the rating if the result ist deleted.

Reply:
Nevermind, I've tested it and you are right. I removed the cascading now!